components/ai_chat/core/browser/utils.cc

```diff
@@ -178,6 +183,65 @@ const std::string& GetActionTypeQuestion(mojom::ActionType action_type) {
   return iter->second;
 }
 
+uint32_t GetMaxPageContentLength(std::optional<uint32_t> context_size) {
+  base::CheckedNumeric<uint32_t> result =
+      context_size.value_or(kCustomModelDefaultContextSize) *
+      kDefaultCharsPerToken;
+  if (!result.IsValid()) {
+    return std::numeric_limits<uint32_t>::max();
+  }
+  return result.ValueOrDie();
+}
+
+uint32_t GetMaxPageContentLength(const mojom::Model& model) {
+  // For Leo models, return the max_page_content_length directly.
+  if (model.options->is_leo_model_options()) {
+    return model.options->get_leo_model_options()->max_page_content_length;
+  }
+
+  // For custom models, delegate to the context size version.
+  return GetMaxPageContentLength(
+      model.options->get_custom_model_options()->context_size);
+}
+
+uint32_t GetLongConversationWarningLength(
+    std::optional<uint32_t> context_size) {
+  base::CheckedNumeric<uint32_t> checked_context_size =
+      context_size.value_or(kCustomModelDefaultContextSize);
```

Comment:
I don't understand what this is doing. `CheckedNumeric` doesn't seem to serve any useful purpose here because the value you're passing in is already a `uint32_t`. Shouldn't this actually be `int32_t` since that is the real max value?

Reply:
also I don't see these methods used outside of tests?